scripts/update_core_lowerbound.py

```diff
@@ -0,0 +1,47 @@
+import requests
+from yaml import safe_load
+from packaging.requirements import Requirement
+from packaging.version import parse
+
+
+core_template_url = "https://raw.githubusercontent.com/pulp/pulpcore/main/template_config.yml"
+
+
+def main():
+    request = requests.get(core_template_url)
+    if request.status_code != 200:
+        print("Failed to find supported branches, not checking lowerbounds")
+        exit(0)
+
+    template = safe_load(request.content)
+    versions = {parse(v) for v in template.get("supported_release_branches", [])}
+    if versions:
+        changed = False
+        with open("requirements.txt") as f:
+            lines = f.readlines()
+            for i, line in enumerate(lines):
+                try:
+                    requirement = Requirement(line.split("#")[0].strip())
+                except ValueError:
+                    pass
+                else:
+                    if requirement.name == "pulpcore":
+                        for spec in requirement.specifier:
+                            if spec.operator == ">=":
+                                min_version = parse(spec.version)
+                                if min_version not in versions:
+                                    # Lowerbound is not a supported branch, modify to next lowest
+                                    valid_lowerbounds = requirement.specifier.filter(versions)
```

Comment:
Yes!!! This was the check I was hoping to find. I think, we should error if no supported branch satisfies the requirement. At that point we can no longer support this branch either.